internal/servers/repository_controller.go

```diff
@@ -0,0 +1,76 @@
+package servers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/servers/store"
+)
+
+func (r *Repository) ListControllers(ctx context.Context, opt ...Option) ([]*store.Controller, error) {
+	return r.listControllersWithReader(ctx, r.reader, opt...)
+}
+
+// listControllersWithReader will return a listing of resources and honor the
+// WithLimit option or the repo defaultLimit. It accepts a reader, allowing it
+// to be used within a transaction or without.
+func (r *Repository) listControllersWithReader(ctx context.Context, reader db.Reader, opt ...Option) ([]*store.Controller, error) {
+	opts := getOpts(opt...)
+	liveness := opts.withLiveness
+	if liveness == 0 {
+		liveness = DefaultLiveness
```

Comment:
Maybe the function description should describe withLiveness since DefaultLiveness has been moved to the worker repo file